src/dmd/e2ir.d

```diff
@@ -5686,6 +5686,18 @@ private elem *toElemStructLit(StructLiteralExp sle, IRState *irs, TOK op, Symbol
         e = el_combine(e, e1);
     }
 
+    if (global.params.useInvariants && sle.sd.inv)
+    {
+        /* If there are no arguments, then this is default initialization and shouldn't call invariant.
+         */
+        if (dim)
```

Comment:
Took me a little bit to work out the reverse logic of that comment and the `if` condition.

Reply:
Yeah. Sth. like "only call invariant for one or more arguments would be easier to read."